features/createroom/test/src/main/kotlin/io/element/android/features/createroom/test/FakeStartDMAction.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.createroom.test
+
+import androidx.compose.runtime.MutableState
+import io.element.android.features.createroom.api.StartDMAction
+import io.element.android.libraries.architecture.Async
+import io.element.android.libraries.matrix.api.core.RoomId
+import io.element.android.libraries.matrix.api.core.UserId
+import io.element.android.libraries.matrix.test.A_ROOM_ID
+import io.element.android.tests.testutils.simulateLongTask
+
+class FakeStartDMAction : StartDMAction {
+
+    private var executeResult: Async<RoomId> = Async.Success(A_ROOM_ID)
+
+    fun givenExecuteResult(result: Async<RoomId>) {
+        executeResult = result
+    }
+
+    override suspend fun execute(userId: UserId, actionState: MutableState<Async<RoomId>>) = simulateLongTask {
+        actionState.value = Async.Loading()
+        actionState.value = executeResult
+    }
```

Comment:
Since there are 2 state value update, I think this is not necessary to use `simulateLongTask`.
Maybe change to
```kotlin
    override suspend fun execute(userId: UserId, actionState: MutableState<Async<RoomId>>) = {
        actionState.value = Async.Loading()
        delay(1)
        actionState.value = executeResult
    }
```
to be closer to the reality? Or create a new `simulateLongTask` which take a `MutableState` in parameter?